app/server/core/repo_analyzer.py

```
@@ -0,0 +1,111 @@
+                    type=EdgeType.CONTAINS
+                )
+                edges.append(edge)
+
+    return RepoStructure(
+        nodes=nodes,
+        edges=edges,
+        root_path=str(repo)
+    )
+
+
+def get_file_nodes(repo_path: str, extensions: List[str] = None) -> List[FileNode]:
```

Ruff (0.14.1)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)